rfqmsg/messages.go

```diff
@@ -182,3 +189,55 @@ type QuoteResponse interface {
 	// String returns a human-readable string representation of the message.
 	String() string
 }
+
+// Uint64FixedPoint is a fixed point record that can be used to encode/decode a
+// fixed point number with a uint64 value to/from a TLV stream.
+type Uint64FixedPoint rfqmath.FixedPoint[rfqmath.GoInt[uint64]]
+
+// NewUint64FixedPoint creates a new fixed point record with a uint64
+// value and a scale.
+func NewUint64FixedPoint(value uint64, scale int) Uint64FixedPoint {
```

Comment:
`scale` here should be a `uint` perhaps? negative values should not be allowed from what I understand

Maybe it should be a `uint8` because that's what it encodes into. Using that type will avoid overflows during encoding.